### PR TITLE
Fix multi lines in content type description

### PR DESCRIPTION
**Pull request by entwicklerfr**
This PR replaces my previous PR : **[WIP] Fix multi lines in content type description #2140**.
Because of a mistake I made on my original forked repo.

My current PR is a 🐛 Bug fix for ISSUE #1543
It's the new PR #2140 which I was not able to merge.

It's a main update on the: Admin

In PR#2140, my reviewer @lauriejim told me:
"With your code, when I insert a new line in the description, I save, then when I try to see my description I lost the multi lines.
The job to do it to not loose it."

Reminder of the ISSUE#1543:
When a user is creating a new Content Type through The Admin/Content Type Builder he can set a description in a text area. If he sets a multilines text

- No saving is possible. an error message is displayed "request.error.model.read"
- If strapi is restarted it doesn't work due to an "SyntaxError: ~/api/MyHappyModel/models/MyHappyModel.settings.json: Unexpected token in JSON at position 152" exception.

The fix consists of updating two files in the strapi-plugin-content-manager

- ContentTypeBuilder.js
- helpers.js

In both `create` and `update model` methods, description is now parsed to replace each newline by a '\n' string.
The replacement is managed by an helper function called `escapeNewlines`.

Multi-lines is kept.

To go further, if I create a `multi4` model type with multilines the json is created as below:
_{
  "connection": "default",
  "collectionName": "multi4",
  "info": {
    "name": "multi4",
    **"description": "multi\nlines\ntext\ntext2\ntext4\n"**
  },
  "options": {
    "timestamps": true
  },
  "attributes": {
    "foo": {
      "default": "",
      "type": "string"
    }
  }
}_

**Admin screenshots in french:**

- Types list
content-type-builder-list

- Multilines type edit 
content-type-builder-edit

### CODE REVIEW

**Comment by entwicklerfr** [comment]
Hi @soupette,

Any feedback ?

File: packages/strapi-plugin-content-type-builder/controllers/ContentTypeBuilder.js [hunk 1]
```diff
@@ -47,11 +48,13 @@ module.exports = {
       return ctx.badRequest(null, [{ messages: attributesErrors }]);
     }
 
+    const _description = escapeNewlines(description, '\\n');
```
Comment (lauriejim) [inline packages/strapi-plugin-content-type-builder/controllers/ContentTypeBuilder.js]: Do we need to store this value in a variable. Just used one time in this function.

File: packages/strapi-plugin-content-type-builder/controllers/ContentTypeBuilder.js [hunk 2]
```diff
@@ -119,8 +124,8 @@ module.exports = {
       modelJSON.connection = connection;
       modelJSON.collectionName = collectionName;
       modelJSON.info = {
-        name,
-        description
+        name: name,
```
Comment (lauriejim) [inline packages/strapi-plugin-content-type-builder/controllers/ContentTypeBuilder.js]: We not need {name: name}

```suggestion
        name,
```

works well.

File: packages/strapi-plugin-content-type-builder/utils/helpers.js [hunk 3]
```diff
@@ -61,10 +61,15 @@ const reorderList = (manager, list) => {
   return List(flattenDeep(reordered));
 };
 
+const escapeNewlines = (content, placeholder = '\n') => {
```
Comment (lauriejim) [inline packages/strapi-plugin-content-type-builder/utils/helpers.js]: Do we need export this function in a helper ?

File: packages/strapi-plugin-content-type-builder/controllers/ContentTypeBuilder.js (hunk 1 above)
Comment (entwicklerfr) [reply packages/strapi-plugin-content-type-builder/controllers/ContentTypeBuilder.js]: I think it's clearer and easier to read an explicit declaration than a direct use of a function, but I can change it if you want. That's not a problem.

File: packages/strapi-plugin-content-type-builder/utils/helpers.js (hunk 3 above)
Comment (entwicklerfr) [reply packages/strapi-plugin-content-type-builder/utils/helpers.js]: Not really. When reading the code I thought it was the right place for this.

**Comment by entwicklerfr** [comment]
U'r welcome.

**Pull request merged**